source/cipher_openssl.c

```diff
@@ -457,12 +468,16 @@ int aws_cryptosdk_sig_sign_start_keygen(
     aws_cryptosdk_sig_abort(*pctx);
     *pctx = NULL;
 
-    aws_string_destroy(*pub_key);
-    *pub_key = NULL;
+    if (pub_key) {
```

Comment:
This fixes a NULL-pointer dereference in the cleanup code.

Reply:
Nice catch